llvm/lib/ProfileData/MemProfReader.cpp

```diff
@@ -633,6 +633,8 @@ Error RawMemProfReader::symbolizeAndFilterStackFrames(
   // Drop the entries where the callstack is empty.
   for (const uint64_t Id : EntriesToErase) {
     StackMap.erase(Id);
+    if(CallstackProfileData[Id].AccessHistogramSize > 0)
```

Comment:
This seems like a bug which could happen elsewhere too. What do you think about adding a destructor to the MemInfoBlock which releases any allocated memory if the histogram size > 0?

Reply:
Yea I was thinking about that too. The problem is you want to keep the two files the same: 

`llvm/include/llvm/ProfileData/MemProfData.inc` and 
`compiler-rt/include/profile/MemProfData.inc`. 

If you add the stuff in the destructor, they will have to diverge because they use different allocators (one with  `malloc`, the other with `InternalAlloc`.

Reply:
Unless we don't mind that they are not the exact same file?

Reply:
No, that makes sense. If we don't free the memory with delete (which is the case in the runtime) then it's not useful. Leaving as it is seems fine.